### PR TITLE
Update Calypso Build to 5.1.0

### PR DESCRIPTION
**Pull request by jeffersonrabb**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR updates `calypso-build` to the latest version, 5.1.0. 

The update removes a few Sass Muriel color variables that the Donate block uses. I didn't want to introduce any style changes in this PR so I created a temporary Sass file with these variables (https://github.com/Automattic/newspack-blocks/compare/update/calypso-build-5.1.0?expand=1#diff-d7cc79577ac5a9dd72abcdbc89faa614). In a future PR we should replace the Muriel color values and remove this file, cc: @thomasguillot. 

The approach to dependency extraction changes in this update to use https://developer.wordpress.org/block-editor/packages/packages-dependency-extraction-webpack-plugin/. 

### How to test the changes in this Pull Request:

1. `npm ci`. Verify `calypso-build` updates to version 5.1.0.
2. `npm run start`. Verify all blocks build with no errors.
3. Test normal operation of all three blocks in editor and view. Should be identical to `master`.
4. Bonus ⭐️: None of the blocks have view-side scripts. Add some Javascript to one  of  the `view.js` files and verify that it executes on the  front end. To verify the dependency enqueuing import a library. E.G.:

```
import classNames from 'classnames';
import { Component } from '@wordpress/element';
console.log( classNames( 'hello', 'classnames' ) );
```

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: class-newspack-blocks.php [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
 	/**
 	 * Enqueue block scripts and styles for editor.
 	 */
 	public static function enqueue_block_editor_assets() {
+		$asset_path = NEWSPACK_BLOCKS__PLUGIN_DIR . NEWSPACK_BLOCKS__BLOCKS_DIRECTORY . 'editor.asset.php';
+		$asset      = file_exists( $asset_path )
+			? require $asset_path
+			: array(
+				'dependencies' => array(),
+				'version'      => '1.0',
```
Comment (retrofox) [inline class-newspack-blocks.php]: sorry for this question, but does it mean this version?

File: class-newspack-blocks.php [hunk 2]
```diff
@@ -108,38 +121,29 @@ public static function enqueue_view_assets( $type ) {
...
 		}
 		if ( file_exists( NEWSPACK_BLOCKS__PLUGIN_DIR . $script_path ) ) {
-			$dependencies = self::dependencies_from_path( NEWSPACK_BLOCKS__PLUGIN_DIR . "dist/{$type}/view.deps.json" );
+			$asset_path = NEWSPACK_BLOCKS__PLUGIN_DIR . "dist/{$type}/view.asset.php";
+			$asset      = file_exists( $asset_path )
+				? require $asset_path
+				: array(
+					'dependencies' => array(),
+					'version'      => '1.0',
+				);
```
Comment (retrofox) [inline class-newspack-blocks.php]: I feel that the removed `dependencies_from_path` method in some way was avoiding duplicated this code. Do you think to keep a method to unify it is a bad idea?

File: class-newspack-blocks.php (hunk 1 above)
Comment (jeffersonrabb) [reply class-newspack-blocks.php]: This is a fallback that theoretically should never be called. In the [documentation example](https://github.com/Automattic/newspack-blocks/pull/232/commits/11c6df0bcac6e6335a7a956e0ed9cd3e4f94ab59) the fallback version uses the script's `filemtime` but  I went for 1.0 just to simplify and make it clear this was a fallback. I could easily be talked out of this 😄

Comment (retrofox) [reply class-newspack-blocks.php]: oh right. I think the `filemtime` was to force to the browser to load the last version of the script

Comment (retrofox) [reply class-newspack-blocks.php]: Oh, the version is automatically defined bu the compiled file.

File: class-newspack-blocks.php (hunk 2 above)
Comment (jeffersonrabb) [reply class-newspack-blocks.php]: This is done in https://github.com/Automattic/newspack-blocks/pull/232/commits/157e179dca52d262d52f20450ec9ad5a9e781224

File: class-newspack-blocks.php (hunk 1 above)
Comment (jeffersonrabb) [reply class-newspack-blocks.php]: Matched the approach shown in the example in https://github.com/Automattic/newspack-blocks/pull/232/commits/157e179dca52d262d52f20450ec9ad5a9e781224

**Comment by jeffersonrabb** [comment]
Thanks for the reviews!

**Pull request merged**